### PR TITLE
chore(rds): add MariaDB engine versions and deprecate old ones

### PR DESCRIPTION
**Pull request by sav-valerio**
#### Taken from [Amazon RDS for MariaDB supports new minor versions 10.6.8, 10.5.16, 10.4.25, 10.3.35, 10.2.44](https://aws.amazon.com/about-aws/whats-new/2022/07/amazon-rds-for-mariadb-supports-new-minor-versions/)
----

### All Submissions:

* [X] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-rds/lib/instance-engine.ts [hunk 1]
```diff
@@ -258,6 +258,10 @@ export class MariaDbEngineVersion {
   public static readonly VER_10_2_40 = MariaDbEngineVersion.of('10.2.40', '10.2');
   /** Version "10.2.41". */
   public static readonly VER_10_2_41 = MariaDbEngineVersion.of('10.2.41', '10.2');
+  /** Version "10.2.43". */
+  public static readonly VER_10_2_43 = MariaDbEngineVersion.of('10.2.43', '10.2');
+  /** Version "10.2.44". */
+  public static readonly VER_10_2_44 = MariaDbEngineVersion.of('10.2.44', '10.2');
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-rds/lib/instance-engine.ts:264]: According to the [MariaDB on Amazon RDS versions](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/MariaDB.Concepts.VersionMgmt.html) page, you can no longer create a new 10.2 instance as of July 15 so I'm not sure we should be adding these here.

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
Thank you for your contribution! Please see my note below about the 10.2 versions. Also, you don't necessarily have to do this in this PR, but I'm thinking we should maybe deprecate the old versions that aren't supported anymore. Thoughts on that?

File: packages/@aws-cdk/aws-rds/lib/instance-engine.ts (hunk 1 above)
Comment (sav-valerio) [reply packages/@aws-cdk/aws-rds/lib/instance-engine.ts]: Fair enough, I missed that note. Removed as you suggested!

**Comment by sav-valerio** [comment]
> Thank you for your contribution! Please see my note below about the 10.2 versions. Also, you don't necessarily have to do this in this PR, but I'm thinking we should maybe deprecate the old versions that aren't supported anymore. Thoughts on that?

I think that it'll be better to do that in a new PR.
Just wondering, can we also remove all together the references to the older versions 10.0 and 10.1 which cannot be launched at all?

**Comment by TheRealAmazonKendra** [comment]
> > Thank you for your contribution! Please see my note below about the 10.2 versions. Also, you don't necessarily have to do this in this PR, but I'm thinking we should maybe deprecate the old versions that aren't supported anymore. Thoughts on that?
> 
> I think that it'll be better to do that in a new PR. Just wondering, can we also remove all references to the older 10.0 and 10.1 versions (along with related minor versions), which cannot be launched at all now?
> 
> https://github.com/aws/aws-cdk/blob/3f01c7daf1fc06c047e6989d7b1d76ab63510f2f/packages/%40aws-cdk/aws-rds/lib/instance-engine.ts#L168
> 
> https://github.com/aws/aws-cdk/blob/3f01c7daf1fc06c047e6989d7b1d76ab63510f2f/packages/%40aws-cdk/aws-rds/lib/instance-engine.ts#L209

I don't think we can just remove them but we should deprecate them.

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
Ah, now I see that we already have these deprecated in a bunch of places. I'm concerned that just removing them would break customer's code, which we don't want to do. I'm also concerned that there are no tests for the MariaDB versions. I know this seems like a very trivial change that shouldn't require tests but I'd like to see some sort of test to ensure we're generating the expected output of these new values. Doesn't need to be 100% coverage or anything.

**Comment by sav-valerio** [comment]
> Ah, now I see that we already have these deprecated in a bunch of places. I'm concerned that just removing them would break customer's code, which we don't want to do. I'm also concerned that there are no tests for the MariaDB versions. I know this seems like a very trivial change that shouldn't require tests but I'd like to see some sort of test to ensure we're generating the expected output of these new values. Doesn't need to be 100% coverage or anything.

@TheRealAmazonKendra I just added the deprecation notices for 10.2.

About the tests, is what you had in mind similar to this?
https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-rds/test/sql-server/sql-server.instance-engine.test.ts#L20

**Comment by TheRealAmazonKendra** [comment]
> > Ah, now I see that we already have these deprecated in a bunch of places. I'm concerned that just removing them would break customer's code, which we don't want to do. I'm also concerned that there are no tests for the MariaDB versions. I know this seems like a very trivial change that shouldn't require tests but I'd like to see some sort of test to ensure we're generating the expected output of these new values. Doesn't need to be 100% coverage or anything.
> 
> @TheRealAmazonKendra I just added the deprecation notices for 10.2.
> 
> About the tests, is this what you had in mind? https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-rds/test/sql-server/sql-server.instance-engine.test.ts#L20

Yeah, something like that is good. I was also thinking of asking for tests on the change to deprecated, but I think that's overkill. We've tested that functionality in general and that would just be extra busy work. Thanks for also adding the EOL notes in there!

**Comment by TheRealAmazonKendra** [comment]
Ugh, the stupid trailing spaces issue. Just run `yarn lint` in this package real quick and it should fix that.

**Review by TheRealAmazonKendra — DISMISSED** [review]
Just putting this back in request changes so I don't keep coming to look at it before change have been made.

**Comment by sav-valerio** [comment]
@TheRealAmazonKendra Tests added for all major versions and fixed the trailing spaces, should be ready now!

Just wondering, as this PR also contains the EOL notices maybe we should change the commit type to something else than _chore_ for making it appear in the changelog?

**Comment by TheRealAmazonKendra** [comment]
> @TheRealAmazonKendra Tests added for all major versions and fixed the trailing spaces, should be ready now!
> 
> Just wondering, as this PR also contains the EOL notices maybe we should change the type to something else other than _chore_ for making it show in the changelog?

I think it's sufficient that it will show up in the doc string.

**Review by TheRealAmazonKendra — APPROVED** [review]

**Pull request merged**